server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -286,7 +292,9 @@ protected void doExecute(Task task, SearchRequest searchRequest, ActionListener<
                 listener
             );
         }
-        executeRequest(task, searchRequest, this::searchAsyncAction, listener);
+        SpanScope scope = tracerFactory.getTracer().startSpan("SearchTask_" + task.getId());
+        TracingActionListener tracingActionListener = new TracingActionListener(tracerFactory, listener, scope);
```

Comment:
TracingAwareListenerWrapper. 
I am not an expert on search code path but do you think instrumenting ```SearchOperationListener``` would easy / clean access to different phases?

Reply:
@shwetathareja I tried implementing SearchOperationListener for instrumenting query and fetch phase but since now we are returning the SpanScope from the startSpan, hence we need to maintain the state (SpanScope returned from the preQueryPhase needs to be cached for ending in the onQueryPhase method.). We can do that in the ThreadLocal since this runs in a single threaded fashion as far as I understood. Need your thoughts on that? @reta